process/common.go

```diff
@@ -276,6 +276,21 @@ func GetMetaHeaderFromPoolWithNonce(
 	return hdr, hash, nil
 }
 
+// GetHeaderFromStorageWithNonce method returns a block header from storage with a given nonce and shardId
+func GetHeaderFromStorageWithNonce(
+	nonce uint64,
+	shardId uint32,
+	storageService dataRetriever.StorageService,
+	uint64Converter typeConverters.Uint64ByteSliceConverter,
+	marshalizer marshal.Marshalizer,
+) (data.HeaderHandler, []byte, error) {
+
+	if shardId == sharding.MetachainShardId {
```

Comment:
This doesn't look right. I know how it is implemented but common.go contains a lot of almost duplicate helper functions. We should have had a common interface of these common funcs with 2 implementations. Might consider this in the next process refactor wave